### PR TITLE
RR-826: Convert goal summary card to component

### PR DESCRIPTION
**Pull request by james-baker-digital-justice**
Converted the goal summary card to a nunjucks component. Added a few unit tests for the bonus features of this approach such as custom classes & attributes as well as the render-if feature for the actions (similar to actions-card). It is otherwise functionally the same but makes some upcoming design changes and known issues with viewing archived goals easier.

### CODE REVIEW

File: server/views/pages/overview/partials/overviewTab/overviewTabPreInduction.njk [hunk 1]
```diff
@@ -22,7 +22,7 @@
         html: createInductionBanner
       }) }}
 
-      {% include '../_goalsList.njk' %}
+      {% include './_goalsList.njk' %}
```
Comment (james-baker-digital-justice) [inline server/views/pages/overview/partials/overviewTab/overviewTabPreInduction.njk:25]: I moved this back to only being shared between the overview screens. My plan is to move the sorting out of the service layer and allow the view to decide the order when iterating

File: server/routes/overview/viewArchivedGoalsController.test.ts [hunk 2]
```diff
@@ -35,14 +35,11 @@ describe('viewArchivedGoalsController', () => {
   })
   it('should get view archived goals page', async () => {
     // Given
-    const expectedTab = 'view-archived-goals'
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/viewArchivedGoalsController.test.ts:38]: The PR in general LGTM 👍  (and I will approve it 😁 ), but the removal of the expected tab from here and the view class itself has made me stop and think....

The `tab` property in the view class (and therefore the expected tab in the tests) is there so that we have a view class for each "tab" on the Overview page.
I'd assumed that the new Archived Goals page was a tab on the Overview page, hence the need for the property etc.
So seeing you remove it here made me look at the designs.
You are right, its not a tab on the Overview page, so you are right to remove the property 👍 

But .... if it's not a tab on the Overview page, and is a page in it's own right, then I think this controller and view class etc are in the wrong folder in the source tree (in that it's nothing to do with the Overview page)

Let's go with this PR as it is 👍 , and then do another one to refactor `viewArchivedGoalsController` and associated classes etc into a better folder structure, outside of the `overview` structure 👍

File: server/views/pages/overview/viewArchivedGoals.njk
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/viewArchivedGoals.njk:1]: Same comment as above - I think this template file is in the wrong folder because the Archived Goals page is not a tab on the Overview page.
But we can move it in a different PR 👍

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

File: server/routes/overview/viewArchivedGoalsController.test.ts (hunk 2 above)
Comment (james-baker-digital-justice) [reply server/routes/overview/viewArchivedGoalsController.test.ts:38]: ok - I left it here on the basis that we're shortly changing the overview and goals/archived/completed will be sub-tabs of the overview. No biggie either way to me.

Comment (nathanrussell-moj-digital) [reply server/routes/overview/viewArchivedGoalsController.test.ts:38]: Ah, in that case, yes. If they will soon become subtabs of the Overview page, then yes, its all in the right place 😁 
Let's leave it for now 😁 👍
